app/src/main/java/org/oppia/android/app/activity/route/ActivityRouter.kt

```diff
@@ -16,22 +17,20 @@ class ActivityRouter @Inject constructor(
   private val consoleLogger: ConsoleLogger
 ) {
   /**  Opens the activity corresponding to the specified [destinationScreen]. */
-  fun routeToScreen(destinationScreen: DestinationScreen) {
-    // TODO: Need to handle error cases properly here.
+  fun routeToScreen(destinationScreen: DestinationScreen): Intent? {
     destinationRoutes[destinationScreen.destinationScreenCase].let { route ->
       when (destinationScreen.destinationScreenCase) {
         DestinationScreenCase.RECENTLY_PLAYED_ACTIVITY_PARAMS -> {
-          activity.startActivity(
-            route?.createIntent(
-              activity,
-              destinationScreen.recentlyPlayedActivityParams
-            )
+          return route?.createIntent(
```

Comment:
This method should be refactored such that nullable isn't possible (since you have a reasonable default case). That should be doable by following an approach like:

```kotlin
return destinationRoutes[...]?.let { route ->
  when (route) {
    ... route is now non-null
  }
} ?: Intent() // Default case.
```

That being said, returning a default Intent seems wrong--what does startActivity() actually do in that case?

Reply:
Yes, you are correct startActivity() works fine and I have updated it.